File: KotlinLibraryX/library/src/main/java/com/lee/library/dialog/WarnDialog.java
package com.lee.library.dialog;

import android.content.Context;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.lee.library.R;
import com.lee.library.dialog.core.BaseAlertDialog;

/**
 * @author jv.lee
 * @date 2020-03-07
 * @description 警告提示dialog
 */
public class WarnDialog extends BaseAlertDialog {

    private String titleText;

    public static WarnDialog build(Context context, String text) {
        WarnDialog dialog = new WarnDialog(context);
        dialog.titleText = text;
        return dialog;
    }

    private WarnDialog(@NonNull Context context) {
        super(context,false);
    }

    @Override
    public int buildViewId() {
        return R.layout.layout_dialog_warm;
    }

    @Override
    public void bindView() {
        ((TextView) findViewById(R.id.tv_title)).setText(titleText == null ? "构建WarnDialog text参数不能为空" : titleText);

        findViewById(R.id.tv_confirm).setOnClickListener(v -> {
            if (confirmListener != null) {
                confirmListener.onConfirm();
            } else {
                dismiss();
            }
        });

    }

}

File: KotlinLibraryX/library/src/main/java/com/lee/library/dialog/WarnDialog.kt
package com.lee.library.dialog

import android.content.Context
import android.view.View
import android.widget.TextView
import com.lee.library.R
import com.lee.library.dialog.core.BaseAlertDialog

/**
 * 警告提示dialog
 * @author jv.lee
 * @date 2020-03-07
 */
class WarnDialog(context: Context) : BaseAlertDialog(context, R.layout.layout_dialog_warn, false) {

    public override fun bindView() {
        findViewById<View>(R.id.tv_confirm).setOnClickListener {
            onConfirm ?: dismiss()
            onConfirm?.invoke()
        }
    }

    override fun setTitle(titleId: Int) {
        findViewById<TextView>(R.id.tv_title).setText(titleId)
    }

    override fun setTitle(title: CharSequence?) {
        findViewById<TextView>(R.id.tv_title).text = title
    }
}